src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,56 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.NotValidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new NotValidDataException("User can't be null");
+        }
+        checkValidAge(user.getAge());
+        checkValidLogin(user.getLogin());
+        checkValidPassword(user.getPassword());
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new NotValidDataException("User with this login already exists");
+        }
+        return storageDao.add(user);
+    }
+
+    private void checkValidLogin(String login) {
+        if (login == null) {
+            throw new NotValidDataException("Login can't be null");
+        }
+        if (login.length() < MIN_LOGIN_LENGTH) {
+            throw new NotValidDataException("Login length must be at least"
+                    + MIN_LOGIN_LENGTH);
+        }
```

Comment:
make your code  lower using one if condition and the same message

Reply:
that's a bad advice, since `null` login and login with inappropriate length are two different things - messages should not be the same